src/actions.rs

```diff
@@ -34,6 +34,7 @@ pub struct InnerAction {
 
     pub merge_infos: Option<MergeInfo>,
     pub merge_needed: bool,
```

Comment:
I can easily see these 2 bools being swapped by mistake at the call site.
Maybe an `Option<ChurnNeeded>` with
```
enum ChurnNeeded {
    Split,
    Merge
}
```
would be safer. It would also capture the fact that they are mutually exclusive, which is true.

Reply:
Yep I agree, type safety for the win!